Fix tests using kubectl convert

masap opened the pull request:
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
```hack/make-rules/test-cmd.sh``` script fails with tariling errors.

```
Error: unknown command "convert" for "kubectl"
```

1. This PR fixes the errors by replacing or removing the use of ```kubectl convert``` option because it was already removed.

2. Fix trailing shell check failure as well.

```
In ./test/cmd/generic-resources.sh line 366:
  kube::test::get_object_assert deployment "{{range.items}}{{$image_field0}}:{{end}}" "${IMAGE_NGINX}:${IMAGE_NGINX}:"
                                                             ^-----------^ SC2154: image_field0 is referenced but not assigned.
```

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs

```

joakimr-axis commented on test/cmd/create.sh:
```diff
@@ -85,7 +85,9 @@ run_kubectl_create_error_tests() {
   # Posting a pod to namespaces should fail.  Also tests --raw forcing the post location
-  grep -q "cannot be handled as a Namespace: converting (v1.Pod)" <<< "$( kubectl convert -f test/fixtures/doc-yaml/admin/limitrange/valid-pod.yaml -o json | kubectl create "${kube_flags[@]}" --raw /api/v1/namespaces -f - --v=8 2>&1 )"
+  kubectl apply "${kube_flags[@]}" -f test/fixtures/doc-yaml/admin/limitrange/valid-pod.yaml
+  grep -q "cannot be handled as a Namespace: converting (v1.Pod)" <<< "$( kubectl get "${kube_flags[@]}" pod/valid-pod -o json | kubectl create "${kube_flags[@]}" --raw /api/v1/namespaces -f - --v=8 2>&1 )"
```
No need for double quotes on static strings. Suggestion:
```shell
grep -q 'cannot be handled as a Namespace: converting (v1.Pod)' <<< "$( kubectl get "${kube_flags[@]}" pod/valid-pod -o json | kubectl create "${kube_flags[@]}" --raw /api/v1/namespaces -f - --v=8 2>&1 )"
```

soltysh replied:
Why not just do:
```bash
kubectl create "${kube_flags[@]}" -f test/fixtures/doc-yaml/admin/limitrange/valid-pod.yaml --raw /api/v1/namespaces --v=8
```
It will give you the exact same effect. Alternatively `kubectl-convert` if you want to stick with convert being used.

masap commented:
@joakimr-axis Thanks! Fixed.

masap commented:
@soltysh Thanks!
As you say, ```kubectl create``` can receive yaml file directly.
But then we get different output from the command like this.

```
$ kubectl create --raw /api/v1/namespaces -f test/fixtures/doc-yaml/admin/limitrange/valid-pod.yaml --v=8
...
I0213 09:48:34.241489  503306 helpers.go:216] server response object: [{
  "metadata": {},
  "status": "Failure",
  "message": "the object provided is unrecognized (must be of type Namespace): couldn't get version/kind; json parse error: invalid character 'a' looking for beginning of value (61706956657273696f6e3a2076310a6b696e643a20506f640a6d65746164 ...)",
  "reason": "BadRequest",
  "code": 400
}]
...
```

This means changing input format to yaml causes changing of test target path.
So I fixed the code to test both of them (json and yaml).

masap commented:
/ok-to-test

soltysh commented on test/cmd/create.sh:
```diff
@@ -85,7 +85,12 @@ run_kubectl_create_error_tests() {
+  # json
+  kubectl apply "${kube_flags[@]}" -f test/fixtures/doc-yaml/admin/limitrange/valid-pod.yaml
+  grep -q 'cannot be handled as a Namespace: converting (v1.Pod)' <<< "$( kubectl get "${kube_flags[@]}" pod/valid-pod -o json | kubectl create "${kube_flags[@]}" --raw /api/v1/namespaces -f - --v=8 2>&1 )"
+  kubectl delete "${kube_flags[@]}" pod/valid-pod
```
You're unnecessarily creating the pod and then removing, just change this so that the get is not needed, b/c it's not. 
Iow:
```bash
kubectl create "${kube_flags[@]}" -f test/fixtures/doc-yaml/admin/limitrange/valid-pod.yaml --raw /api/v1/namespaces --v=8
```
will be sufficient b/c you'll try to create a pod against endpoint for namespaces which will return you just that error.

masap replied:
I see. Fixed.

masap committed:
Fix tests using kubectl convert

hack/make-rules/test-cmd.sh script fails with tariling errors.

Error: unknown command "convert" for "kubectl"

1. This PR fixes the errors by replacing or removing the use of
"kubectl convert" option because it was already removed.

2. Fix trailing shell check failure as well.
In ./test/cmd/generic-resources.sh line 366:
  kube::test::get_object_assert deployment "{{range.items}}{{$image_field0}}:{{end}}" "${IMAGE_NGINX}:${IMAGE_NGINX}:"

soltysh reviewed (approved):
/lgtm
/approve
